### PR TITLE
Fix build script for generation

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR addresses a failure in the build scripts for this repo. It now
ensures complete generation of artifacts from the source protos at build
time. This should enable detection of out-of-date artifacts in presubmits.

As a consequence of this PR, several problematic artifacts are updated.

Most importantly, the outdated authentication protos are updated to reflect
the new `networking/v1alpha3` protos (instead of the deleted `routing/v1alpha2`
protos.

This PR also updates the circleci build image to `go1.10`.

### CODE REVIEW

File: authentication/v1alpha1/istio.authentication.v1alpha1.pb.html [hunk 1]
```diff
@@ -181,7 +181,7 @@ <h2 id="Policy">Policy</h2>
   name: mTLS-enable
   namespace: frod
 spec:
-  match:
+  destinations:
```
[the remark this answers is not in the document]
Comment (douglas-reid) [reply authentication/v1alpha1/istio.authentication.v1alpha1.pb.html]: the comments didn't match the protos. this fixes that issue.

File: Makefile [hunk 2]
```diff
@@ -160,7 +161,9 @@ generate-broker-go: $(broker_v1_pb_gos) $(broker_v1_pb_doc)
 
 $(broker_v1_pb_gos) $(broker_v1_pb_doc): $(broker_v1_protos) | depend $(protoc_gen_go) $(protoc_bin)
 	## Generate broker/dev/*.pb.go + $(broker_v1_pb_doc)
-	@$(protoc) $(proto_path) $(protoc_gen_go_plugin) $(protoc_gen_docs_plugin)$(broker_v1_path) $^
+	ls -l vendor/github.com/gogo
```
Comment (geeknoid) [inline Makefile]: Are these two lines here on purpose, or leftover debugging?

File: Makefile [hunk 3]
```diff
@@ -97,6 +97,7 @@ ifeq ($(UNAME), Linux)
 	sudo mv protoc3/include/* /usr/local/include/
 	rm -f protoc-3.5.0-linux-x86_64.zip
 	rm -rf protoc3
+	ls -R
```
Comment (geeknoid) [inline Makefile]: Leftover debugging?

**Comment by douglas-reid** [comment]
OK. This PR now ensures circleci gates the PRs properly. Prow needs to be updated to go1.10.

File: Makefile (hunk 3 above)
Comment (douglas-reid) [reply Makefile]: ongoing debugging.

**Comment by douglas-reid** [comment]
PTAL. I think this will resolve short-term generation check issues with istio/api.

I'll follow up on a different channel for prow, etc.

**Comment by ldemailly** [comment]
merging as head is currently broken and this fixes it

**Pull request merged**